data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileFlowCnecCreator.java

```diff
@@ -0,0 +1,390 @@
+            return false;
+        }
+        return true;
+    }
+
+    private PropertyBags getAssessedElementsWithContingencies(String assessedElementId, PropertyBag assessedElementPropertyBag) {
+        PropertyBags assessedElementsWithContingencies = CsaProfileCracUtils.getLinkedPropertyBags(assessedElementsWithContingenciesPropertyBags, assessedElementPropertyBag, CsaProfileConstants.REQUEST_ASSESSED_ELEMENT, CsaProfileConstants.REQUEST_ASSESSED_ELEMENT);
+
+        if (assessedElementsWithContingencies.isEmpty()) {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCOMPLETE_DATA, "no link between the assessed element and a contingency"));
+        }
+        return assessedElementsWithContingencies;
+    }
+
+    private PropertyBags getCurrentLimits(String assessedElementId, PropertyBag assessedElementPropertyBag) {
+        PropertyBags currentLimits = CsaProfileCracUtils.getLinkedPropertyBags(currentLimitsPropertyBags, assessedElementPropertyBag, CsaProfileConstants.REQUEST_CURRENT_LIMIT, CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_OPERATIONAL_LIMIT);
+
+        if (currentLimits.isEmpty()) {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCOMPLETE_DATA, "no current limit linked with the assessed element"));
+        }
+        return currentLimits;
+    }
+
+    private Set<Contingency> checkLinkAssessedElementContingency(String assessedElementId, PropertyBag assessedElementWithContingencies, Set<Contingency> combinableContingenciesSet, Boolean isCombinableWithContingency) {
+        Set<Contingency> combinableContingencies = combinableContingenciesSet.stream().collect(Collectors.toSet());
+        String normalEnabledWithContingencies = assessedElementWithContingencies.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_WITH_CONTINGENCY_NORMAL_ENABLED);
+
+        if (normalEnabledWithContingencies != null && !Boolean.parseBoolean(normalEnabledWithContingencies)) {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.NOT_FOR_RAO, "AssessedElementWithContingency.normalEnabled is false"));
+            return null;
+        }
+
+        String contingencyId = assessedElementWithContingencies.getId(CsaProfileConstants.REQUEST_CONTINGENCY);
+        String combinationConstraintKind = assessedElementWithContingencies.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_WITH_CONTINGENCY_COMBINATION_CONSTRAINT_KIND);
+        if (CsaProfileConstants.ASSESSED_ELEMENT_WITH_CONTINGENCIES_LINK_CONSIDERED.equals(combinationConstraintKind)) {
+            csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.NOT_FOR_RAO, "AssessedElementWithContingency.combinationConstraintKind is considered"));
```

Comment:
ImportStatus.inconstincency rather than not for RAO since it's in entsoe spec for all AE profiles

Reply:
done